src/main/java/com/powsybl/openloadflow/network/PiModelArray.java

```diff
@@ -218,9 +218,10 @@ public Optional<Direction> updateTapPositionR1(double deltaR1, int maxTapShift,
         int oldTapPosition = tapPosition;
         // find tap position with the closest r1 value without exceeding the maximum of taps to switch.
         double smallestDistance = Math.abs(deltaR1);
-        for (int p = positionRange.getMinimum();
-             p < positionRange.getMaximum() && Math.abs(lowTapPosition + p - oldTapPosition) <= maxTapShift;
-             p++) {
+        for (int p = positionRange.getMinimum(); p < positionRange.getMaximum(); p++) {
```

Comment:
I am not sure to get the difference here. Can you explain me with words?

Reply:
I think that in the first case, if (Math.abs(lowTapPosition + p - oldTapPosition) <= maxTapShift) is false, then the loop is stopped. In the second case, the loop goes to next iterate.

In case p goes from 0 to (oldTapPosition - lowTapPosition) and lowTapPosition and oldTapPosition are distant by more than maxTapShift, then the loop starts at p=0 and directly stops (in the first case) because the condition (Math.abs(lowTapPosition + p - oldTapPosition) <= maxTapShift) is false.

Reply:
Yes, agreed.